src/planner/vectorfield/VectorFieldUtil.cpp

```diff
@@ -115,19 +116,22 @@ bool computeJointVelocityFromTwist(
     if (position + _timestep * velocityLowerLimit
         < positionLowerLimit + _padding)
       velocityLowerLimits[i] = std::max(
-          velocityLowerLimits[i],
-          ((velocityLowerLimits[i] + _padding) - position) / _timestep);
+          velocityLowerLimit,
+          ((positionLowerLimit + _padding) - position) / _timestep);
+    initialGuess[i] = std::max(velocityLowerLimits[i], initialGuess[i]);
 
     if (position + _timestep * velocityUpperLimit
         > positionUpperLimit - _padding)
       velocityUpperLimits[i] = std::min(
-          velocityUpperLimits[i],
-          ((velocityUpperLimits[i] - _padding) - position) / _timestep);
+          velocityUpperLimit,
+          ((positionUpperLimit - _padding) - position) / _timestep);
+    initialGuess[i] = std::min(velocityUpperLimits[i], initialGuess[i]);
```

Comment:
Maybe we can use `std::clamp` instead?

Reply:
`std::clamp` seems a feature since `c++17`.